window_test.go

```diff
@@ -112,6 +112,34 @@ func TestStartGreaterThanEnd(t *testing.T) {
 	assert.Equal(t, time.Duration(-1), w.UntilNextInterval(interval))
 }
 
+func TestMorningToMorning(t *testing.T) {
+	// Window not active just between 10am and 11am each day.
+	w := window.New(mkTime(11, 0), mkTime(10, 0))
+
+	w.Now = mkNow(9, 59)
+	assert.True(t, w.Active())
+	assert.Equal(t, time.Duration(0), w.Until())
+	assert.Equal(t, time.Minute, w.UntilEnd())
+
+	w.Now = mkNow(10, 0)
+	assert.False(t, w.Active())
+	assert.Equal(t, time.Hour, w.Until())
+
+	w.Now = mkNow(10, 59)
+	assert.False(t, w.Active())
+	assert.Equal(t, time.Minute, w.Until())
+
+	w.Now = mkNow(11, 0)
+	assert.True(t, w.Active())
+	assert.Equal(t, time.Duration(0), w.Until())
+	assert.Equal(t, 23*time.Hour, w.UntilEnd())
+
+	w.Now = mkNow(17, 0)
+	assert.True(t, w.Active())
+	assert.Equal(t, time.Duration(0), w.Until())
+	assert.Equal(t, 17*time.Hour, w.UntilEnd())
+}
```

Comment:
Just a minor thing and it doesn't need to be fixed.  But for another person reading the test, it would probably It probably would have been better to choose another time than 17:00 since the 17 hours till end also is just luck that they match.  It could be confusing to someone or possibly hide a bug.     eg.   18:00, would mean 16 hours for w.UntilEnd().

Reply:
Good call. I'll fix.